elearning-backend/app/Http/Controllers/API/AdminCategoriesController.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace App\Http\Controllers\API;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\CreateCategoriesPostRequest;
+use App\Http\Requests\UpdateCategoriesPostRequest;
+use App\Models\Category;
+use Illuminate\Http\Request;
+use Illuminate\Validation\Rule;
+
+class AdminCategoriesController extends Controller
+{
+    // Fetch all categories
+    public function index()
+    {
+        $categories = Category::all();
+
+        return response()->json([
+            'categories'=> $categories,
+        ]);
+    }
+
+    // Get selected category
+    public function getCategory(Request $request)
+    {
+        $category = Category::where("id", $request->id)->get()->first();
+
+        return response()->json([
+            'category'=> $category,
+        ]);
+    }
```

Comment:
since you use `/categories/{id}`, utilize the id then you can find directly the category since id is unique
```suggestion
    public function getCategory($id)
    {
        $category = Category::find($id);

        return response()->json($category );
    }
```
you will use ->json([...]) only if you have multiple returned data

Reply:
Changes have been made to code